skt/executable.py

```diff
@@ -321,8 +321,32 @@ def cmd_run(cfg):
     """
     global retcode
     runner = skt.runner.getrunner(*cfg.get('runner'))
-    retcode = runner.run(cfg.get('buildurl'), cfg.get('krelease'),
-                         cfg.get('wait'), uid=cfg.get('uid'))
+    state = read_state()
+
+    # Ensure that the state file has a krelease (comes from build step)
+    if not state.get('krelease'):
+        print(
+            "ERROR:"
+            "The 'krelease' item is missing from skt's state file. Please "
+            "run `skt build` before attempting to test a kernel."
+        )
+        sys.exit(1)
+
+    # Ensure that the state file has a buildurl (comes from the publish step)
+    if not state.get('buildurl'):
+        print(
+            "ERROR:"
+            "The 'buildurl' item is missing from skt's state file. Please "
+            "run `skt publish` before attempting to test a kernel."
+        )
+        sys.exit(1)
```

Comment:
This will definitely help the user.

However, I would like this solved more efficiently. Instead of checking for
the presence of every variable in the state file, how about we have the state
recorded and verified explicitly?

We can e.g. have state "none" for nothing done, "merged" for code retrieved,
"built" for kernel built, and "tested" for kernel tested. Executing any of the
corresponding commands could then check that the state is the previous one or
above, and reset it accordingly.

Do you think you have time to implement the above in another PR? If not, then
let's not add these error messages lest they proliferate and increase our code
size.

Also, adding error messages like these would be better done in a separate
commit. I.e. refactor for the new interface in one, add functionality in
another.